app/code/Magento/PageCache/etc/varnish6.vcl

```diff
@@ -148,65 +148,44 @@ sub vcl_hash {
 }
 
 sub vcl_backend_response {
-
+	# Serve stale content for three days after object expiration
+	# Perform asynchronous revalidation while stale content is served
     set beresp.grace = 3d;
 
+    # All text-based content can be parsed as ESI
     if (beresp.http.content-type ~ "text") {
         set beresp.do_esi = true;
     }
 
+    # Allow GZIP compression on all JavaScript files and all text-based content
     if (bereq.url ~ "\.js$" || beresp.http.content-type ~ "text") {
         set beresp.do_gzip = true;
     }
 
+    # Add debug headers
     if (beresp.http.X-Magento-Debug) {
         set beresp.http.X-Magento-Cache-Control = beresp.http.Cache-Control;
     }
 
-    # cache only successfully responses and 404s that are not marked as private
-    if (beresp.status != 200 &&
-            beresp.status != 404 &&
-            beresp.http.Cache-Control ~ "private") {
+    # Only cache HTTP 200 and HTTP 404 responses
+    if (beresp.status != 200 && beresp.status != 404) {
```

Comment:
@toonvd Do we want to cache 301 & 302 redirects?

Reply:
Good remark, I usually refrain from caching them since it causes confusion for developers / end users. (especially 302)

Reply:
perhaps @damienwebdev and/or @danslo have an opinion on this